UnityProject/Assets/Scripts/Managers/TimedEventsManager.cs

```diff
@@ -10,26 +11,62 @@ namespace Managers
 	/// Manager that handles timed game events that only happen under a specific time of the year/month/week/day
 	/// </summary>
 	/// TODO : Allow admins to create and save events for their sever
-	public class TimedEventsManager : SingletonManager<TimedEventsManager>
+	public class TimedEventsManager : SingletonManager<TimedEventsManager>, IInitialise
 	{
 		[SerializeField] private List<TimedGameEventSO> events;
 		private List<TimedGameEventSO> activeEvents = new List<TimedGameEventSO>();
 
 		public List<TimedGameEventSO> ActiveEvents => activeEvents;
 
+		public InitialisationSystems Subsystem { get; }
+
+		public void Initialise()
+		{
+			Logger.Log("[Subsystems/TimedEvents] - Setting up event hooks.");
+			EventManager.AddHandler(Event.RoundStarted, StartActiveEvents);
+			EventManager.AddHandler(Event.ScenesLoadedServer, CleanAndUpdateActiveEvents);
+			EventManager.AddHandler(Event.RoundEnded, EndActiveEvents);
+		}
 
 		public override void Awake()
 		{
 			base.Awake();
+			//Update on awake so the UI can see what events are there.
 			UpdateActiveEvents();
 		}
 
-		public void UpdateActiveEvents()
+		private void CleanAndUpdateActiveEvents()
+		{
+			Logger.Log("[SubSystems/TimedEvents] - Cleaning active events.");
+			UpdateActiveEvents();
+		}
+
+		private void StartActiveEvents()
+		{
+			Logger.Log("[Subsystems/TimedEvents] - Starting timed events.");
+			foreach (var timedEvent in activeEvents)
+			{
+				StartCoroutine(timedEvent.EventStart());
+			}
+		}
+
+		private void EndActiveEvents()
+		{
+			Logger.Log("[Subsystems/TimedEvents] - Stopping timed events.");
+			foreach (var timedEvent in activeEvents)
+			{
+				StartCoroutine(timedEvent.OnRoundEnd());
```

Comment:
Because this is called when the subsystems load, we need access to `WaitFor` to set up things after a frame has passed or to await other coroutines and such.

Reply:
Even when the round ends?

Reply:
> Even when the round ends?

Just in case you have like hundreds of things that you want to do without freezing round end.
For example, scanning an entire matrix for objects with a specific component.

Reply:
Fair enough.